### PR TITLE
Scanpy update

### PR DESCRIPTION
**Pull request by bebatut**
FOR CONTRIBUTOR:
* [X] I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [X] License permits unrestricted use (educational + commercial)
* [ ] This PR adds a new tool or tool collection
* [X] This PR updates an existing tool or tool collection
* [ ] This PR does something else (explain below)

---

With this PR, it will be possible to run scanpy tutorials:
- [Clustering 3K PBMCs](https://scanpy-tutorials.readthedocs.io/en/latest/pbmc3k.html)
- [PAGA for hematopoiesis in mouse](https://scanpy-tutorials.readthedocs.io/en/latest/paga-paul15.html)
- [Cell cycle](https://github.com/theislab/scanpy_usage/blob/master/180209_cell_cycle/cell_cycle.ipynb)

I expanded the possibilities of Anndata tool to inspect and manipulate Anndata datasets. Some missing functions of scanpy were added. And the outputs now have the function name displayed (to make it easier to follow of the steps).

TODO: update tests

### CODE REVIEW

**Comment by bebatut** [comment]
Ready for reviews :smile:

- All tests have been updated
- Functions have been reorganized between the wrappers
- Missing functions have been added

With this version, almost all functions of scanpy should be there

File: tools/scanpy/inspect.xml [hunk 1]
```diff
@@ -37,143 +255,647 @@ adata.obs.to_csv('$obs', sep='\t')
...
+                <has_text_matching expression="only_positive=True"/>
+            </assert_stdout>
+            <output name="anndata_out" file="tl.rank_genes_groups.liblinear.krumsiek11.h5ad" ftype="h5ad" compare="sim_size">
+                <assert_contents>
+                    <has_h5_keys keys="X, obs, obsm, raw.X, raw.var, uns, var" />
                 </assert_contents>
             </output>
-            <output name="obs" file="tl.dpt.diffmap.neighbors_gauss_braycurtis.recipe_weinreb17.paul15_subsample.obs.tabular" compare="sim_size"/>
+        </test>
+        <!--<test>
```
[the remark this answers is not in the document]
Comment (bebatut) [reply tools/scanpy/inspect.xml]: The corresponding function `tl.marker_gene_overlap` does not really work with the current `scanpy` version. I implemented it before testing it. I keep it ready for the next release of scanpy (which will hopefully fix that)

File: tools/scanpy/cluster_reduce_dimension.xml [hunk 2]
```diff
@@ -1,32 +1,46 @@
-<tool id="scanpy_cluster_reduce_dimension" name="Cluster and reduce dimension with scanpy" version="@galaxy_version@">
+<tool id="scanpy_cluster_reduce_dimension" name="Cluster, embed and infer trajectories with scanpy" version="@galaxy_version@">
```
[the remark this answers is not in the document]
Comment (bebatut) [reply tools/scanpy/cluster_reduce_dimension.xml]: I changed the names of all wrappers

File: tools/anndata/manipulate.xml [hunk 3]
```diff
@@ -211,6 +402,10 @@ The possible manipulations are:
 
     Data matrix is transposed, observations and variables are interchanged.
 
+- Add annotation (TODO)
```
[the remark this answers is not in the document]
Comment (bebatut) [reply tools/anndata/manipulate.xml]: Thanks for pointing that. It is implemented, I updated the documentation

File: tools/anndata/manipulate.xml [hunk 4]
```diff
@@ -177,6 +314,60 @@ adata.write('anndata.h5ad')
...
+                <param name="var_obs" value="var"/>
+                <conditional name="filter">
+                    <param name="filter" value="index"/>
+                    <conditional name="index">
+                        <param name="format" value="text"/>
+                        <param name="text" value="Gata2,EKLF"/>
+                    </conditional>
+                </conditional>
+            </conditional>
+            <output name="anndata" value="manipulate.filter_var_index.h5ad" ftype="h5ad" compare="sim_size"/>
```
Comment (bgruening) [inline tools/anndata/manipulate.xml]: You could now use the new HDF asserts in newer Galaxy's. https://docs.galaxyproject.org/en/latest/dev/schema.html#tool-tests-test-output-assert-contents

Comment (bebatut) [reply tools/anndata/manipulate.xml]: I know but sometimes the changes are not obvious and I was a bit lazy to open each generated h5ad and check exactly the changes

Comment (bebatut) [reply tools/anndata/manipulate.xml]: Should I change to use mostly the HDF asserts?

**Comment by bgruening** [comment]
Aweomse!

**Pull request merged**